app/src/main/java/com/hedvig/app/feature/home/ui/claimstatus/composables/ClaimStatusCard.kt

```diff
@@ -24,7 +25,11 @@ fun ClaimStatusCard(
     uiState: ClaimStatusCardUiState,
     modifier: Modifier = Modifier,
     isClickable: Boolean = false,
+    onClaimCardShown: (String) -> Unit,
 ) {
+    LaunchedEffect(uiState) {
+        onClaimCardShown(uiState.id)
+    }
```

Comment:
This will now send a new event in case some part of the UI changes, like if the UI is pulled to refresh, and the status changed from in progress to paid for example. Is this something we want to track with `hAnalytics.claimCardVisible(claimId, claim.status.rawValue)`?
Intuitively, I am thinking that the "card shown" event should just have a `Unit` key in LaunchedEffect.
[iOS](https://github.com/HedvigInsurance/ugglan/blob/34aef0933c8e13766adf0f1bde70098c3f8b6b9b/Projects/Home/Sources/Claims/ClaimStatus.swift#L20-L25) seem to trigger this on `trackOnAppear` which triggers once [when the view first appears](https://developer.apple.com/documentation/SwiftUI/AnyView/onAppear(perform:)) so we may want to do the same or tell them to ask it as well?

Reply:
Good point, perhaps we could key it to `uiState.id`? That is after all the relevant part 🤔

Reply:
Yeah that's possible as well.
As I am thinking about it now, I just wonder now how would this function since the cards are all shown in a row of items using a pager, so they're lazily shown 🤔.
If a new claim is inserted into the row, and their index is moved around, will the existing items get fed the new data, aka change the ID aka send the event again?
I am afraid the more I think about it the less I am certain about how it actually functions.
Unless you have a better understanding how the lazy list would work in these scenarios, maybe just using id as a key is good enough. I think no matter what we do this event trigger a bit more often than what we'd probably prefer it to.